pkg/query/api/v1.go

```diff
@@ -288,18 +290,24 @@ func (api *API) query(r *http.Request) (interface{}, []error, *ApiError) {
 		warnmtx.Unlock()
 	}
 
+	raw := r.FormValue("query")
+	level.Debug(api.logger).Log("query", raw)
+
 	// We are starting promQL tracing span here, because we have no control over promQL code.
 	span, ctx := tracing.StartSpan(r.Context(), "promql_instant_query")
+	span.SetTag("query", raw)
 	defer span.Finish()
 
 	begin := api.now()
-	qry, err := api.queryEngine.NewInstantQuery(api.queryableCreate(enableDedup, 0, enablePartialResponse, warningReporter), r.FormValue("query"), ts)
+	qry, err := api.queryEngine.NewInstantQuery(api.queryableCreate(enableDedup, 0, enablePartialResponse, warningReporter), raw, ts)
 	if err != nil {
+		level.Warn(api.logger).Log("query", raw, "error", errorBadData)
```

Comment:
I'm not so sure about this one. This will fail when someone sends an invalid query, which will happen a lot and this will pollute the logs.